offchain-modules/src/apps/rpc/handler.ts

```diff
@@ -0,0 +1,194 @@
+      //   const assetType = getAssetTypeByAsset(payload.asset.ident.address);
+      //   let unsignedTx;
+      //   switch (assetType) {
+      //     case 'trx':
+      //       unsignedTx = await tronWeb.transactionBuilder.sendTrx(committee, amount, sender);
+      //       break;
+      //     case 'trc10':
+      //       unsignedTx = await tronWeb.transactionBuilder.sendToken(
+      //         committee,
+      //         amount,
+      //         payload.asset.ident.address,
+      //         sender,
+      //       );
+      //       break;
+      //     case 'trc20':
+      //       const options = {};
+      //       const functionSelector = 'transfer(address,uint256)';
+      //       const params = [
+      //         { type: 'address', value: committee },
+      //         { type: 'uint256', value: amount },
+      //       ];
+      //       unsignedTx = await tronWeb.transactionBuilder.triggerSmartContract(
+      //         payload.asset.ident.address,
+      //         functionSelector,
+      //         options,
+      //         params,
+      //         sender,
+      //       );
+      //       break;
+      //     default:
+      //       Promise.reject(new Error('invalid tron asset type'));
+      //   }
+      //   const memo = recipientLockscript.toAddress().toCKBAddress().concat(',').concat('sudt extra data');
+      //   tx = await tronWeb.transactionBuilder.addUpdateData(unsignedTx, memo, 'utf8');
+      default:
+        Promise.reject(new Error('invalid chain type'));
+    }
+    const bridgeFee = {
+      network: network,
+      ident: { address: payload.asset.ident.address },
+      amount: '1',
+    };
+    return {
+      network: network,
+      rawTransaction: tx,
+      bridgeFee: bridgeFee,
+    };
+  }
+
+  async generateBridgeOutNervosTransaction(
+    payload: API.GenerateBridgeOutNervosTransactionPayload<EthereumNetwork>,
+  ): Promise<API.GenerateTransactionResponse<NervosNetwork>> {
+    console.log('generateBridgeOutNervosTransaction ', payload);
+    const fromLockscript = Script.fromRPC({
+      code_hash: payload.sender.codeHash,
+      args: payload.sender.args,
+      hash_type: payload.sender.hashType,
+    });
+    const ownLockHash = ForceBridgeCore.ckb.utils.scriptToHash(<CKBComponents.Script>fromLockscript);
+
+    const network = payload.network;
+    const assetName = payload.asset.ident.address;
+
+    let asset;
+    switch (network) {
+      case 'Ethereum':
+        asset = new EthAsset(assetName, ownLockHash);
+        break;
+      // case 'Tron':
+      //   asset = new TronAsset(assetName, ownLockHash);
+      //   break;
+      default:
+        Promise.reject(new Error('invalid chain type'));
+    }
+
+    const amount = payload.asset.amount;
+
+    const ckbTxGenerator = new CkbTxGenerator(ForceBridgeCore.ckb, new IndexerCollector(ForceBridgeCore.indexer));
+    const burnTx = await ckbTxGenerator.burn(fromLockscript, payload.recipient.address, asset, new Amount(amount));
```

Comment:
the amount is `AmountWithoutDecimal` , we should be careful when using PW `Amount`

Reply:
```amount``` here identifies sudt amount, not ckb amount

Reply:
But when we `new Amount(1)` without second param, the `Amount` is  is actually `Amount { amount: '100000000' }
`, are you sure we don't need to set the second argument `decimal` to 0? Can `ckbTxGenerator.burn` handle such an amount correctly?